scripts/Muon/GUI/Common/thread_model_wrapper.py

```diff
@@ -26,8 +26,12 @@ def execute(self):
 class ThreadModelWrapperWithOutput(object):
     def __init__(self, function=None):
         self.callback = function
+        self.args = None
         self.result = None
 
+    def set_args(self, *args):
```

Comment:
Can we simplify this and have everything done in 1 step in `__init__`? Args will be an empty tuple if nothing is passed?

```python
def __init__(self, function, *args):
        self.callback = function
        self.args = args
        self.result = None
```

It looks like the function would be better not to be optional here as it's not checked later?

For calling, an empty tuple will then expand to nothing with `*` so calling will be

```python
    def execute(self):
        self.result = self.callback(*self.args)
```

Reply:
Thanks, that is much better!